lms/resources/_js_config.py

```diff
@@ -73,35 +73,34 @@ def enable_content_item_selection_mode(self, form_action, form_fields):
             HTML form that we'll use to submit the user's chosen document
         :type form_fields: dict
         """
-        self._config.update(
-            {
-                "mode": "content-item-selection",
-                "enableLmsFilePicker": False,
-                "formAction": form_action,
-                "formFields": form_fields,
-                "googleClientId": self._request.registry.settings["google_client_id"],
-                "googleDeveloperKey": self._request.registry.settings[
-                    "google_developer_key"
-                ],
-                # Pass the URL of the LMS that is launching us to our JavaScript code.
-                # When we're being launched in an iframe within the LMS our JavaScript
-                # needs to pass this URL (which is the URL of the top-most page) to Google
-                # Picker, otherwise Picker refuses to launch inside an iframe.
-                "customCanvasApiDomain": self._context.custom_canvas_api_domain,
-                "lmsUrl": self._context.lms_url,
+        self._config["mode"] = "content-item-selection"
+
+        def google_picker_origin():
+            # Pass the URL of the LMS that is launching us to our JavaScript code.
+            # When we're being launched in an iframe within the LMS our JavaScript
+            # needs to pass this URL (which is the URL of the top-most page) to Google
+            # Picker, otherwise Picker refuses to launch inside an iframe.
+            return self._context.custom_canvas_api_domain or self._context.lms_url
+
+        self._config["filePicker"] = {
+            "formAction": form_action,
+            "formFields": form_fields,
```

Comment:
The roughed out version (https://github.com/hypothesis/lms/issues/1435#issuecomment-598738736) had these as `action` and `fields` as the form part seems like an implementation detail.

Reply:
I'd like to keep the `form` prefix here: these are config settings that the backend is sending to the frontend for the frontend to use as the _form_ action and the _form_ fields